modules/madvertiseBidAdapter.js

```diff
@@ -0,0 +1,131 @@
+import * as utils from 'src/utils';
+import {registerBidder} from 'src/adapters/bidderFactory';
+
+// use protocol relative urls for http or https
+const MADVERTISE_ENDPOINT = 'https://mobile.mng-ads.com/';
+
+export const spec = {
+  code: 'madvertise',
+  aliases: ['madvertise'],
+  /**
+   * @param {object} bid
+   * @return boolean
+   */
+  isBidRequestValid: function (bid) {
+    if (typeof bid.params !== 'object') {
+      return false;
+    }
+    if (!Array.isArray(bid.sizes) || bid.sizes.length == 0) {
```

Comment:
we have a utility function to parse the bid.sizes. See https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L114